doc/differentiation/autodiff.qbk

```diff
@@ -0,0 +1,805 @@
+[/          Copyright Matthew Pulver 2018 - 2019.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)]
+
+[/ DO NOT EDIT THIS FILE - THIS IS AUTOMATICALLY GENERATED FROM DOXYGEN OUTPUT USING html2qbk.rb. ]
+
+[section:autodiff Automatic Differentiation]
+[template autodiff_equation[name]  '''<inlinemediaobject><imageobject><imagedata fileref="../equations/autodiff/'''[name]'''"></imagedata></imageobject></inlinemediaobject>''']
+
+[h1:synopsis 
+Synopsis]
+
+ 
+    #include <boost/math/differentiation/autodiff.hpp>
+    
+    namespace boost { namespace math { namespace differentiation { namespace autodiff {
+    
+    // The primary template alias for instantiating autodiff variables.
+    template<typename RealType,size_t Order,size_t... Orders>
+    using variable = typename nested_dimensions<RealType,Order,Orders...>::type;
+    
+    // Satisfies Boost's Conceptual Requirements for Real Number Types.
+    // Don't use this dimension<> class directly. Instead, use the variable<> alias.
+    template<typename RealType,size_t Order>
+    class dimension
+    {
+      public:
+    
+        // Initialize a variable of differentiation.
+        explicit dimension(const root_type&);
+    
+        // The root data type chosen when variable<> is declared. E.g. float, double, etc.
+        explicit operator root_type() const;
+    
+        // Query return value of computation to get the derivatives.
+        template<typename... Orders>
+        typename type_at<RealType,sizeof...(Orders)-1>::type derivative(Orders... orders) const;
+    
+        // All of the arithmetic operators are overloaded.
+        template<typename RealType2,size_t Order2>
+        dimension<RealType,Order>& operator+=(const dimension<RealType2,Order2>&);
+    
+        dimension<RealType,Order>& operator+=(const root_type&);
+    
+        // ...
+    };
+    
+    // Standard math functions are overloaded and called via argument-dependent lookup (ADL).
+    template<typename RealType,size_t Order>
+    dimension<RealType,Order> floor(const dimension<RealType,Order>&);
+    
+    template<typename RealType,size_t Order>
+    dimension<RealType,Order> exp(const dimension<RealType,Order>&);
+    
+    // ...
+    
+    } } } } // namespace boost::math::differentiation::autodiff
+    
+
+[h1:description 
+Description]
+
+Autodiff is a header-only C++ library that facilitates the [@https://en.wikipedia.org/wiki/Automatic_differentiation automatic differentiation] (forward mode) of mathematical functions in single and multiple variables.
```

Comment:
> Since you're incorporating this into boost.math, I don't think calling this a library is the correct language.

Can you suggest an alternative? I've been thinking about this and nothing more appropriate comes to mind. API? C++ header files?

IMO I think calling it a library is appropriate. Compare to the first sentence in the [Statistical Functions and Distributions](https://www.boost.org/doc/libs/1_69_0/libs/math/doc/html/math_toolkit/stat_tut/overview/headers.html) "library" documentation:

> All the code in this library is inside namespace boost::math.

This is in reference to the sub-library of just these functions, which itself is a library.

Reply:
It's a nitpick; let's leave it as is.